### ISSUE TITLE
"Find references" doesn't find overloads

### ISSUE THREAD

**Issue by UnboundVariable** [astral-sh/ty#886]
### Summary

When using the "go to references" or "find all references" feature in the language server, if you point to the name of an overloaded function or method, I would expect all overloads to be found in addition to the overload's implementation. Currently, only the implementation is found.

**Comment by InSyncWithFoo** [comment]
Don't you mean, when `textDocument/references` is triggered on an overload's function name, ty should only return references <em>specific</em> to that overload?

```python
@overload
def f() -> None: ...
@overload
def f(v: str) -> None: ...
#   ^ Case 1: Overload function name
def f(v: str = '') -> None: ...
#   ^ Case 2: Implementation function name

f()       # Only 2
f('foo')  # Both 1 and 2
```

**Comment by AdemFr** [comment]
Not sure if also relevant / similar issue: https://github.com/astral-sh/ty/issues/2362

**Comment by Hugo-Polloli** [comment]
https://github.com/astral-sh/ruff/pull/23012 fixes the matching side by adding `navigation_targets_intersect` (so multi-target symbols compare correctly once discovered)
For this issue, I think the remaining work is to expand the initial seed definitions for overload declarations (include overload group + implementation)
Also the shape should be similar to the defs expansion implemented by https://github.com/astral-sh/ruff/pull/22999/changes

**Comment by Hugo-Polloli** [comment]
I'm working on the fix right now
